### ISSUE TITLE
Update Multi-select component

### ISSUE THREAD

**Issue by i-am-chitti** [Travelopia/web-components#15]
The current multi-select has a few bugs and needs some enhancements which are as follows -
- [ ] Skip disabled options during keyboard navigation

https://github.com/Travelopia/web-components/assets/60139930/6c05c2e3-f186-420c-a07d-98aed1bb0114

- [ ] Bring option into view if options container is scrollable
- [ ] Initialize multi-select component with selected options - Currently if a `tp-multi-select-option` is provided `selected="yes"` from code on first page load, the status and pills are not showing the selected values.
 
https://github.com/Travelopia/web-components/assets/60139930/6bc364ef-f1f8-4bae-8a67-d16d12f1f2ef

- [ ] On using a single select inside a form, the first value is submitted by default if there is no empty option (whose value is not present). 

https://github.com/Travelopia/web-components/assets/60139930/a5d4ea7d-9e3e-4c70-8f76-243feee4fa78

cc: @junaidbhura

**Comment by i-am-chitti** [comment]
@junaidbhura Need your re-opinion on this -

- [ ] Showing "No results found" message on library user side has some development overhead. If there is no matching search, all options, groups should be hidden. This is provided from library side itself, just group hiding has to be taken care of by implementor. Now, in order to assert that there is no option result from search, the library implementer has to check if all options has 'hidden===yes` attribute. Can we have an attribute on `tp-multi-select` which is set to "yes" if there is no search result so that implementer has a better source of truth ?

**Comment by junaidbhura** [comment]
Thanks @i-am-chitti I've added some notes on the PR. Regarding the empty groups, can this be achieved using CSS? https://developer.mozilla.org/en-US/docs/Web/CSS/:empty

**Comment by i-am-chitti** [comment]
@junaidbhura I found one more bug that 'change' event is being dispatched even while calling `select`. This is causing conflict with the usage, though not sure if we should have it. Here is an use-case -
I'm setting the selected option after an AJAX call. The options are selected programatically through `select` method available on `tp-multi-select`. On doing so, all those handlers that are listening to 'change' are triggered. But this ain't a **user provoked select event**. Consider this with respect to the trip search form on Exodus where there are two multi-select and one autocomplete. Now, the activity select change is listened by date select. While activity selection is done programatically during state initialization from local storage and via autocomplete. For every call to `select` function, an API call is made. Now suppose, a user selected all. When the page is refreshed and during restoring the select states, we would be calling `select` method for every selected option. Now, for every call, there is an API call made as `change` is fired. This leads to the same number of API calls as number of selected options. So, I think we should only dispatch the events when user triggers them.

In the draft PR, I've refactored the code to dispatch events only when the user triggers them. This also requires us to call `update` of component manually. I saw the codebase of 'Choices JS' specially there `set` Let me know if there is any better way to resolve this.
